modules/gdscript/language_server/gdscript_extend_parser.cpp

```diff
@@ -269,7 +269,20 @@ void ExtendGDScriptParser::parse_class_symbol(const GDScriptParser::ClassNode *p
 					if (j > 0) {
 						symbol.detail += ", ";
 					}
-					symbol.detail += m.signal->parameters[j]->identifier->name;
+					const ParameterNode &this_parameter = *m.signal->parameters[j];
+					symbol.detail += this_parameter.identifier->name;
+					if (this_parameter.datatype_specifier && this_parameter.datatype_specifier->type_chain.size() == 1) {
+						const Variant::Type type = GDScriptParser::get_builtin_type(this_parameter.datatype_specifier->type_chain[0]->name);
+
+						// todo: If in `GDScriptParser::parse_signal()` it can be assumed that the type is always a built-in type then the following test is not required.
```

Comment:
```suggestion
						// TODO: If in `GDScriptParser::parse_signal()` it can be assumed that the type is always a built-in type, then the following test is not required.
```